logic/src/classes/Station.java

```diff
@@ -9,52 +9,39 @@
 
     private Coordinate coordinate;
     private String name;
-    private Map<Station, Road> stationsFromCurrent2Roads;
-    private Map<Station, Road> stationsToCurrent2Roads;
-
-    private Set<Station> allStationReachableFromThisStation = null;
-
-    public Station(Coordinate coordinate, String name) {
-        this.coordinate = coordinate;
-        this.name = name;
-
-        this.stationsFromCurrent2Roads = new HashMap<>();
-        this.stationsToCurrent2Roads = new HashMap<>();
+    private Map<Station, Road> stationsFromCurrent;
+    private Map<Station, Road> stationsToCurrent;
+    private Set<Station> allReachableStationFromThisStation;
+
+    public Station(Coordinate sationCoordinate, String stationName) {
+        coordinate = sationCoordinate;
+        name = stationName;
+        stationsFromCurrent = new HashMap<>();
+        stationsToCurrent = new HashMap<>();
+        allReachableStationFromThisStation =  new HashSet<>();
     }
 
     public Road getRoadToStation(Station station){
-        return stationsFromCurrent2Roads.get(station);
+        return stationsFromCurrent.get(station);
     }
 
     public void addRoadFromCurrentStation(Road road) {
         // A , input:A->B :  A.roadsFrom += (B, A->B)   , B.roadTo += (A, A->B)
         //TODO: add validation for start_station == current Station
-        //TODO: support two side road
-        Station thisStation = road.getStartStation();
-        Station targetStation = road.getEndStation();
-        this.stationsFromCurrent2Roads.put(targetStation, road);
-        targetStation.stationsToCurrent2Roads.put(thisStation, road);
-    }
-
-    private Set<Station> getAllReachableStations(){
-        Set<Station> allReachableStation = new HashSet<>(this.stationsFromCurrent2Roads.keySet());
-        this.stationsFromCurrent2Roads.keySet().forEach((s)-> allReachableStation.addAll(s.getAllReachableStations()));
+        //TODO: support two side road לא אמרנו שנעשה את זה פשוט לכל כיוון?
+        Station fromStation = road.getStartStation();
+        Station totStation = road.getEndStation();
 
-        return allReachableStation;
+        stationsFromCurrent.put(totStation, road);
+        allReachableStationFromThisStation.add(totStation);
```

Comment:
but if you are station A and there is station B which can reach stations C, D, E, F, G
then you add the Road(A->B)
int your case you are updating A that it can reach now station B but it's not correct - you are missing all the C, D, E ,F, G which are now reachable from A :(
The function getAllReachableStations created just to avoid this use case.